### ISSUE TITLE
navigate to worksheet by key and gid

### ISSUE THREAD

**Issue by pita2k** [burnash/gspread#271]
is it possible to add a feature that will return a worksheet class by key and by gid ?

**Comment by msuozzo** [comment]
@m4rc1e if you want to take a crack at this, please patch away.

**Comment by m4rc1e** [comment]
@msuozzo I'll have it sorted soon.

**Comment by wolverdude** [comment]
Whatever happened to this? It would be super useful to get a reference to a worksheet by copy-pasting the URL, and this feature is a prerequisite to that.

**Comment by burnash** [comment]
@wolverdude could you please share an example usage for this / your use case scenario?

**Comment by wolverdude** [comment]
What I would like is something like

```py
gc.get_worksheet_by_url('https://docs.google.com/spreadsheet/ccc#gid=0Bm...FE
```

This would return a specific tab from a spreadsheet. I feel like that’s a per common use-case, just wanting some specific data from a specific sheet, and it’s easy to just copy-paste the url.

This issue would be a perquisite to that, and at least halfway there. I envision the API would be something like:

```py
sh.get_worksheet _by_id(1234567890)
```

**Comment by wolverdude** [comment]
I’d be happy to put in a PR for it.

**Comment by burnash** [comment]
Great, the PR is welcome!

**Comment by lavigne958** [comment]
closed by #857
